debug-typescript.sh

```
@@ -0,0 +1,268 @@
+    else
+        echo -e "${RED}TypeScript logs directory not found${NC}"
+    fi
+}
+
+# Function to analyze project structure for potential issues
+analyze_project_structure() {
+    echo -e "${YELLOW}🏗️ Analyzing Project Structure...${NC}"
+    
+    # Check for large files that might cause issues
+    echo -e "${BLUE}Large TypeScript files (>100KB):${NC}"
+    find . -name "*.ts" -o -name "*.tsx" -o -name "*.js" -o -name "*.jsx" | xargs ls -la | awk '$5 > 100000 {print $9, $5}' | sort -k2 -nr
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)